packages/fuel-indexer-database/postgres/src/lib.rs

```diff
@@ -554,6 +554,31 @@ pub async fn latest_assets_for_index(
     })
 }
 
+pub async fn last_block_height_for_indexer(
+    conn: &mut PoolConnection<Postgres>,
+    namespace: &str,
+    identifier: &str,
+) -> sqlx::Result<u64> {
+    #[cfg(feature = "metrics")]
+    METRICS
+        .db
+        .postgres
+        .last_block_height_for_indexer_calls
+        .inc();
+
+    let query = format!(
+        "SELECT MAX(id) FROM {namespace}_{identifier}.indexmetadataentity LIMIT 1"
+    );
+
+    match sqlx::query(&query).fetch_one(conn).await {
+        Ok(row) => {
+            let id: i64 = row.get(0);
+            Ok(id as u64)
+        }
+        Err(_) => Ok(0),
```

Comment:
Block heights will never be `0`. If we're keeping this logic we should make this `1`